metals/src/main/scala/scala/meta/internal/metals/JavaInteractiveSemanticdb.scala

```diff
@@ -0,0 +1,227 @@
+package scala.meta.internal.metals
+
+import java.nio.file.Files
+import java.nio.file.Path
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.util.Try
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.MD5
+import scala.meta.internal.process.SystemProcess
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.io.AbsolutePath
+
+import com.typesafe.config.ConfigFactory
+import com.typesafe.config.ConfigParseOptions
+import com.typesafe.config.ConfigSyntax
+
+class JavaInteractiveSemanticdb(
+    javac: AbsolutePath,
+    jdkVersion: JavaInteractiveSemanticdb.JdkVersion,
+    pluginJars: List[Path],
+    workspace: AbsolutePath,
+    buildTargets: BuildTargets
+) {
+
+  private val readonly = workspace.resolve(Directories.readonly)
+
+  def textDocument(source: AbsolutePath, text: String): s.TextDocument = {
+    val sourceRoot = source.parent
+    val workDir = AbsolutePath(
+      Files.createTempDirectory("metals-javac-semanticdb")
+    )
+    val targetRoot = workDir.resolve("target")
+
+    val targetClasspath = buildTargets
+      .inferBuildTarget(sourceRoot)
+      .flatMap(buildTargets.targetJarClasspath)
+      .getOrElse(Nil)
+      .map(_.toString)
+
+    val jigsawOptions = addExportsFlags ++ patchModuleFlags(source, sourceRoot)
+    val mainOptions =
+      List(
+        javac.toString,
+        "-cp",
+        (pluginJars ++ targetClasspath).mkString(":"),
+        "-d",
+        targetRoot.toString,
+        s"-Xplugin:semanticdb -sourceroot:${sourceRoot} -targetroot:${targetRoot}"
+      )
+    val cmd = mainOptions ::: jigsawOptions ::: source.toString :: Nil
+
+    val stdout = List.newBuilder[String]
+    val ps = SystemProcess.run(
+      cmd,
+      workDir,
+      false,
+      Map.empty,
+      outLine => stdout += outLine,
+      errLine => stdout += errLine
+    )
+
+    val future = ps.complete.recover { case NonFatal(e) =>
+      scribe.error("Running javac-semanticdb failed", e)
+      1
+    }
+
+    val exitCode = Await.result(future, 10.seconds)
```

Comment:
There would be no need to wait if we returned future from the method.

Reply:
yep. but it's will trigger much-much more changes. `InteractiveSemanticdb` does the same thing for PC because of `Semandticdb` interface signature.
I can turn it into an issue

Reply:
It's fine, we anyway want to limit the time we spend on the compilation.